lib/ModelMixins/ReferenceMixin.ts

```diff
@@ -135,6 +136,30 @@ function ReferenceMixin<T extends AbstractConstructor<BaseType>>(Base: T) {
       return result;
     }
 
+    /**
+     * Recursively load a nested chain of references till the given maxDepth.
+     *
+     * If this reference points to another reference and so on.. then this
+     * method will load them all up to the given depth.
+     *
+     * @param maxDepth The maximum depth up to which the references should be resolved.
+     * @returns A promise that is fulfilled with a Result value when all the references have been loaded.
+     */
+    async recursivelyLoadReference(maxDepth: number): Promise<Result<void>> {
+      let currentTarget: BaseModel | undefined = this;
+      const errors: TerriaError[] = [];
+      while (maxDepth > 0 && ReferenceMixin.isMixedInto(currentTarget)) {
+        (await currentTarget.loadReference()).pushErrorTo(errors);
+        currentTarget = currentTarget.target;
+        maxDepth -= 1;
+      }
+      const result = TerriaError.combine(
+        errors,
+        "Failed to recursively load reference `${this.uniqueId}`"
+      );
+      return Result.error(result);
```

Comment:
```suggestion
      const maybeError = TerriaError.combine(
        errors,
        "Failed to recursively load reference `${this.uniqueId}`"
      );
      return Result.none(maybeError);
```

Use `Result.none` to convey that the result is normally an empty successful result. I'm not sure if the name `maybeError` is a bit too haskell'ish.

Reply:
Good catch @steve9164!

I have changed it to the suggested pattern.

`Result.none` sounds a bit counter-intuitive to me though - from the name it sounds like it should always return an empty `Result`. Maybe it should be called `errorOrNone`?

Reply:
It's not the best name. At some point I'd like to refactor Result a bit. I'll make a ticket and add that point